kernel/src/schema/mod.rs

```diff
@@ -458,6 +458,73 @@ impl StructType {
     }
 }
 
+impl IntoIterator for StructType {
+    type Item = StructField;
+    type IntoIter = StructFieldIntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StructFieldIntoIter {
+            inner: self.fields.into_values(),
+        }
+    }
+}
+
+impl<'a> IntoIterator for &'a StructType {
+    type Item = &'a StructField;
+    type IntoIter = StructFieldRefIter<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StructFieldRefIter {
+            inner: self.fields.values(),
+        }
+    }
+}
+
+pub struct StructFieldIntoIter {
+    inner: indexmap::map::IntoValues<String, StructField>,
+}
+
+impl Iterator for StructFieldIntoIter {
+    type Item = StructField;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.inner.size_hint()
+    }
+}
+
+impl ExactSizeIterator for StructFieldIntoIter {
+    fn len(&self) -> usize {
+        self.inner.len()
+    }
+}
+
+/// Iterator that yields references to StructFields
+pub struct StructFieldRefIter<'a> {
```

Comment:
can we derive `Debug` for both pub types and probably derive `Clone` for the Ref one? (i am pretty sure `Values` already implements clone)